### PR TITLE
Laboratory work #4, Lopatko Yulia - 21FPL1

### PR DESCRIPTION
**Pull request by lazykosti**

### CODE REVIEW

**Comment by artyomtugaryov** [comment]
nothing to review(

**Review by artyomtugaryov — APPROVED** [review]
What about 6,8,10?

File: lab_4_summarization_textrank/main.py [hunk 1]
```diff
@@ -2,15 +2,33 @@
...
+from typing import Union, Any, Type
+import re
 from lab_3_keywords_textrank.main import TextEncoder, \
     TextPreprocessor
 
 PreprocessedSentence = tuple[str, ...]
 EncodedSentence = tuple[int, ...]
 
 
+def check_type(user_input: Any, correct_type: Type) -> bool:
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:14]: why do you return True?  I don't understand why you need the return value if in a bad case you raise the Error

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_4_summarization_textrank/main.py (hunk 1 above)
Comment (lazykosti) [reply lab_4_summarization_textrank/main.py:14]: you mean I can just return an **error** if I put **not** in a condition and that's it?

Comment (lazykosti) [reply lab_4_summarization_textrank/main.py:14]: but returning **True** helps me with going on with commands. Returning **error** allows me not to write it every time the function returns **False**

File: lab_4_summarization_textrank/main.py [hunk 2]
```diff
@@ -84,59 +108,79 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
         :param text: the raw text
         :return: a sequence of sentences
         """
-        pass
+        check_type(text, str)
+        list_of_sentences = []
+        text = text.replace('\n', ' ').replace('  ', ' ')
+        sentences = re.split(r'(?<=[.!?])\s+(?=[A-ZА-Я])', text)
+        for number, sentence in enumerate(sentences):
+            if sentence:
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:129]: no nesting

File: lab_4_summarization_textrank/main.py [hunk 3]
```diff
@@ -84,59 +108,79 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
 
     def _preprocess_sentences(self, sentences: tuple[Sentence, ...]) -> None:
         """
         Enriches the instances of sentences with their preprocessed versions
         :param sentences: a list of sentences
         :return:
         """
-        pass
+        check_sequence(sentences, tuple, Sentence)
+        if sentences:
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:140]: no nested

File: lab_4_summarization_textrank/main.py [hunk 4]
```diff
@@ -84,59 +108,79 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
-
     def _learn_indices(self, tokens: tuple[str, ...]) -> None:
         """
         Fills attributes mapping words and integer equivalents to each other
         :param tokens: a sequence of string tokens
         :return:
         """
-        pass
+        check_sequence(tokens, tuple, str)
+        start = 1000
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:168]: own attribute of instances

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -146,7 +190,13 @@ def calculate_similarity(sequence: Union[list, tuple], other_sequence: Union[lis
     :param other_sequence: a sequence of items
     :return: similarity score
     """
-    pass
+    if not isinstance(sequence, (list, tuple)) or not isinstance(other_sequence, (list, tuple)):
+        raise ValueError
+    if not sequence or not other_sequence:
+        return 0
+    one_set = set(sequence)
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:197]: ?

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -184,7 +240,18 @@ def add_edge(self, vertex1: Sentence, vertex2: Sentence) -> None:
...
         """
-        pass
+        check_type(vertex1, Sentence)
+        check_type(vertex2, Sentence)
+        if vertex1 == vertex2:
+            raise ValueError
+        for one_vertex in vertex1, vertex2:
+            if one_vertex not in self._vertices:
+                self._vertices.append(one_vertex)
+                row = [calculate_similarity(one_vertex.get_encoded(), other_vertex.get_encoded())
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:250]: duplicated computations?

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**